src/backends/sentry_backend_crashpad.cpp

```diff
@@ -149,17 +149,19 @@ sentry__crashpad_handler(int signum, siginfo_t *info, ucontext_t *user_context)
 #    endif
 
             SENTRY_TRACE("invoking `on_crash` hook");
-            should_handle
-                = options->on_crash_func(&uctx, options->on_crash_data);
+            should_dump = options->on_crash_func(&uctx, options->on_crash_data);
         } else if (options->before_send_func) {
             sentry_value_t event = sentry_value_new_event();
             SENTRY_TRACE("invoking `before_send` hook");
             event = options->before_send_func(
                 event, nullptr, options->before_send_data);
+            should_dump = !sentry_value_is_null(event);
```

Comment:
This was missing up to now. For the crashpad backend, it was irrelevant whether you returned the event or a null value.

The other difference (which I have not changed) to the other backends: we create the event right before the hook and then discard it again. The other backends allow changes made to the event received in `before_send()` to land in the backend.